ui/quick_query_panel.py

```diff
@@ -200,3 +216,89 @@ def show_query(self):
             self.dialog.text_query.setPlainText(query)
             item = self.dialog.menu_widget.item(self.dialog.query_menu_index)
             self.dialog.menu_widget.setCurrentItem(item)
+
+            
+    def update_friendly(self):
+        """ Updates the QuickQuery Friendly Label (label_qq_friendly) """
+
+        # Although self.gather_values() might be a tempting way to fetch
+        # the required info, this calls super.nominatim_value() which
+        # executes file I/O - this crashes X
+        # Therefore query the widget values directly
+
+        query_type = self.dialog.query_type_buttons[self.panel].currentData()
+        place = self.dialog.places_edits[self.panel].text()
+        if place is not None:
+            place = place.strip()
+            if place == '':
+                place = None
+        distance = self.dialog.spin_place_qq.value()
+        layerobj = self.dialog.layers_buttons[self.panel].currentLayer()
+        layer = None if (layerobj is None) else layerobj.name()
+            
+        # first translate the location information
+        have_loc_info = False
+        if query_type == 'in':
+            if place is not None:
+                geomsg = "within " + place
+                have_loc_info = True
+                
+        elif query_type == 'around':
+            if place is not None:
+                geomsg = tr("within ") + str(distance)
+                geomsg = geomsg + tr(" meters of ") + place
+                have_loc_info = True
+
+        elif query_type == 'canvas':
+            geomsg = tr(" in the canvas extent ")
+            have_loc_info = True
+        elif query_type == 'layer':
+            if layer is not None:
+                geomsg = tr(" in the layer extent '") + layer + "' "
```

Comment:
About these strings:
I'm trying to use what they call the "new way" of formatting strings in Python: https://pyformat.info/ especially 'Named placeholders'. It's much better in translation system. According to languages, the order of words may change etc ...

The best approach, not easy, would have a few sentences with tokens such as for the all values one:
ALL_VALUES = 'All OSM objects with the key {key} in {extent} are going to be downloaded' so `extent` can be replaced with list of cities, or `name layer extent` or `canvas extent`.